File: CoreEngine/DevModule/tiles/Stone/Stone.rb
include CoreEngine::Entities
include CoreEngine::Scripting

class Stone < MapTile
	def	register
		super
		self.SetEntityProperties({ draw: true, update: true })
		self.DrawOnLayer(2, :second_draw)
		p "Registering 1"
	end

	def create
		super
		@iterations = 0
	end

	def draw
		super
		Canvas.draw_string(CoreFont.System, @iterations.to_s, position.x, position.y, CoreColor.White)
	end

	def update
		@iterations += 1
	end

	def second_draw
		Canvas.DrawFilledRect(position.x, position.y, 4, 4, CoreColor.Black)
	end
end

File: DevModule/Tiles/Stone.cs
﻿using PlumeAPI.Graphics;
using Microsoft.Xna.Framework;
using PlumeRPG.Entities;
using PlumeAPI.Networking;
using PlumeAPI.Networking.Builtin;
using System;
using PlumeAPI.Entities;

namespace DevModule.Tiles {
	public class Stone : MapTile {
		Random random;
		public Stone() : base() {
		}

		public int Bazoople { get; set; }

		public Stone(int x, int y) : base(x, y) {
			DrawOnLayer(2, "SecondDraw");
			SetEntityProperty("update", true);
			Bazoople = 0;
			random = new Random(("" + Position.X + Position.Y).GetHashCode());
			RegisterProperty(EntityPropertyType.Syncable, "Bazoople", () => { return Bazoople; }, (value) => { Bazoople = (int)value; });
		}

		public override void UpdateServer() {
			if(random.Next(0,100) > 80) Bazoople++;
		}

		public override void Draw() {
			base.Draw();
			Canvas.DrawString(FontRepository.System, "" + Bazoople, (int) Position.X, (int) Position.Y, Color.White);
		}
		public void SecondDraw() {
			Canvas.DrawFilledRect((int) Position.X, (int) Position.Y, 4, 4, Color.Black);
		}
	}
}
